dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/data/JdbcAnalyticsManager.java

```diff
@@ -180,6 +182,10 @@ public Future<Map<String, Object>> getAggregatedDataValues(
         params = queryPlanner.assignPartitionsFromQueryPeriods(params, tableType);
       }
 
+      if (params.hasSubexpressions() && params.getSubexpression().hasPeriodOffsets()) {
```

Comment:
Makes me wonder if this now introduces the feature of sub-expressions to the particular expression language used here or if it always had subexpressions already.

Reply:
I'm not sure I understand the question. We had subExpressions before (for indicator expressions only). They just didn't support periodOffset until this PR.

Reply:
Is this change now also for indicator expressions or some other type of expression so we basically adding the feature of subExpressions to that language. That was my question.

Reply:
Ah, thanks. subExpressions are still only for indicator expressions.